### DISCUSSION TITLE
In Stylet IoC, can I add in new types at runtime after IoC container has been configured?

### DISCUSSION THREAD

**Discussion by mwdpb** [canton7/Stylet#386] [Q&A] [answered]
Typically the known type bindings happen in the `ConfigureIoC` method, this works fine when you know all types statically.  In my case, I have some types that are only known at runtime and I would like to add bindings for them using reflection.  Can I do that?  In another word, is the IoC container only allowed to be built once or it can be added more types and built again at later time?

**Comment by canton7** [comment, marked as answer]
It doesn't look like there's a method to create a builder from a container. You can keep a reference to the builder around, though, and keep adding to it, then create another container from it later.

**Reply by mwdpb** [reply]
Thanks for the quick answer, keeping a reference to the builder does the trick!  Thanks!

**Reply by liulz2024** [reply]
I am a beginner, can you write an example? Thank you!
